### PR TITLE
feat: register

### PR DESCRIPTION
**Pull request by vikingpooh**
- add userservice
- add usercontroller
- add userrouter
- add register functionality

### CODE REVIEW

File: server/controllers/userController/register.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import {body} from 'express-validator';
+import httpStatus from 'http-status';
+import { MESSAGES } from 'consts';
+import { userService } from 'services';
+import { ArgumentValidationError, CustomError } from 'errors';
+import { errorHandlerWrapper } from 'utils/errorHandler.wrapper';
+import { Logger, encryptPassword } from 'utils';
+
+export const registerValidator = () => {
+    console.log("registered");
```
Comment (genie099) [inline server/controllers/userController/register.ts]: remove console log

File: server/controllers/userController/register.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+};
+
+type Params = unknown;
+type ResBody = unknown;
+type ReqBody = {
+    email?: string;
+    first_name?: string;
+    last_name?: string;
+    password1: string;
+    password2: string;
```
Comment (genie099) [inline server/controllers/userController/register.ts]: what is the password1 and password2?

Comment (vikingpooh) [reply server/controllers/userController/register.ts]: password and confirm password.

File: server/controllers/userController/register.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+}
+type ReqQuery = unknown;
+
+export const registerHandler =async (
+    req: Request<Params, ResBody, ReqBody, ReqQuery>,
+    res: Response
+) => {
+    const {email, first_name, last_name, password1, password2} = req.body;
```
Comment (genie099) [inline server/controllers/userController/register.ts]: please beautify. add space right after opening bracket and right before closing bracket.

File: server/entities/User.entity.ts [hunk 4]
```diff
@@ -6,27 +6,18 @@ import { CoreEntity } from "./Core.entity";
...
 
-  @Column({ name: "display_name", nullable: true })
-  displayName?: string;
-
-  @Column({ name: "avatar", nullable: true })
-  avatar?: string;
-
-  @Column({ name: "notification_setting", length: 3 })
-  notificationSetting: string;
-
```
Comment (genie099) [inline server/entities/User.entity.ts]: this entity is different from db design.

File: server/controllers/userController/register.ts (hunk 3 above)
Comment (vikingpooh) [reply server/controllers/userController/register.ts]: Yes. Got it.

**Review by genie099 — COMMENTED** [review]
@TelmanWisley I left some comments in this PR. please take a look.

File: server/controllers/userController/register.ts (hunk 2 above)
Comment (genie099) [reply server/controllers/userController/register.ts]: checking password and confirm password should be done in the frontend side.

Comment (genie099) [reply server/controllers/userController/register.ts]: @TelmanWisley

**Review by genie099 — APPROVED** [review]

**Pull request merged**